pkg/server/api/agent/v1/service.go

```diff
@@ -434,11 +434,9 @@ func (s *Service) CreateJoinToken(ctx context.Context, req *agentv1.CreateJoinTo
 
 	expiry := time.Now().Unix() + int64(req.Ttl)
 
-	result, err := s.ds.CreateJoinToken(ctx, &datastore.CreateJoinTokenRequest{
-		JoinToken: &datastore.JoinToken{
-			Token:  req.Token,
-			Expiry: expiry,
-		},
+	err = s.ds.CreateJoinToken(ctx, &datastore.JoinToken{
+		Token:  req.Token,
+		Expiry: time.Unix(expiry, 0),
```

Comment:
We should improve this to use the clock provided to the service. I think we can also do this without going in and out of "unix" time. The former should allow this behavior to be tested deterministically.
```suggestion
		Expiry: s.clk.Now().Add(time.Duration(req.Ttl) * time.Second),
```

Reply:
I did this in handler, which gets us out of using Unix() back and forth
```
time.Now().Add(time.Second * time.Duration(request.Ttl))
```